### PR TITLE
[#2346] improvement(docs): Add configurations items about tree lock

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add configuration descriptions about the usage of tree lock. 

### Why are the changes needed?

To improve docs. 

Fix: #2346 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

N/A.

### CODE REVIEW

File: docs/gravitino-server-config.md [hunk 1]
```diff
@@ -63,6 +63,17 @@ You can also specify filter parameters by setting configuration entries of the f
 We strongly recommend that you change the default value of `gravitino.entity.store.kv.rocksdbPath`, as it's under the deployment directory and future version upgrades may remove it.
 :::
 
+### Tree lock configuration
+
+Gravitino server uses tree lock to ensure the consistency of the data. The tree lock is a memory lock (Currently, Gravitino only supports in memory lock) that can be used to ensure the consistency of the data in Gravitino server. for more impormation about tree lock, please refer to design [doc1](https://docs.google.com/document/d/1fAxwWdgOHNouysJNgilI5WbCzF1MiTVqsaiGiUga5VA/edit#heading=h.f8rd8hp69vwq) and [doc2](https://docs.google.com/document/d/1ov81tnkRAUkac-K8AjfTsK2AAbpCfnSBfE9XqrTlyWc/edit).
```
Comment (jerryshao) [inline docs/gravitino-server-config.md:68]: Remove the design doc here, don't put the design doc into the user doc, it is unrelated.

Comment (yuqi1129) [reply docs/gravitino-server-config.md]: done

File: docs/gravitino-server-config.md [hunk 2]
```diff
@@ -63,6 +63,16 @@ You can also specify filter parameters by setting configuration entries of the f
...
+### Tree lock configuration
+
+Gravitino server uses tree lock to ensure the consistency of the data. The tree lock is a memory lock (Currently, Gravitino only supports in memory lock) that can be used to ensure the consistency of the data in Gravitino server. The configuration items are as follows:
+
+| Configuration item                   | Description                                                   | Default value | Since Version |
+|--------------------------------------|---------------------------------------------------------------|---------------|---------------|
+| `gravitino.lock.maxNodes`            | The maximum number of tree lock nodes to keep in memory       | 100000        | 0.5.0         |
+| `gravitino.lock.minNodes`            | The minimum number of tree lock nodes to keep in memory       | 1000          | 0.5.0         |
+| `gravitino.lock.cleanIntervalInSecs` | The interval in seconds to clean up the stale tree lock nodes | 60            | 0.5.0         |
+
```
Comment (jerryshao) [inline docs/gravitino-server-config.md:75]: I think you miss one column `required`.

Comment (yuqi1129) [reply docs/gravitino-server-config.md:75]: Sorry, I copied the format from block `Catalog configuration` and miss something.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**